# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/ipc/GfxMessageUtils.h
# Commit: 71c122ac0ca7
# Full Hash: 71c122ac0ca73391866b1ef19f4f82bc2d28568b
# Author: David Parks <davidp99@gmail.com>
# Date: 2020-01-09 03:40:18
# Regressor Bug: 1477756
# File Overlap Count: 1
# Description:
#   Bug 1477756 - Initial out-of-process WebGL implementation. r=mccr8,handyman
#   
#   Splits WebGLContext into ClientWebGLContext and HostWebGLContext.  The Client enables the JS-control of a WebGL context in a content procecss while the Host executes the WebGL graphics operations (via a WebGLContext that maintains much of the existing code) in the compositor process.  At this point, the cross-process behavior is disabled -- this series of patches is an incremental step toward that final goal.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D54018
# ==============================================================================

diff -r 5cbf5bb11c58 -r 71c122ac0ca7 gfx/ipc/GfxMessageUtils.h
--- a/gfx/ipc/GfxMessageUtils.h	Wed Jan 08 21:52:03 2020 +0000
+++ b/gfx/ipc/GfxMessageUtils.h	Wed Jan 08 22:19:14 2020 +0000
@@ -37,6 +37,7 @@
 namespace mozilla {
 
 typedef gfxImageFormat PixelFormat;
+enum WebGLVersion : uint32_t { WEBGL1, WEBGL2 };
 
 }  // namespace mozilla
 
@@ -1179,6 +1180,13 @@
     : public BitFlagsEnumSerializer<mozilla::SideBits,
                                     mozilla::SideBits::eAll> {};
 
+template <>
+struct ParamTraits<mozilla::WebGLVersion>
+    : public ContiguousEnumSerializerInclusive<mozilla::WebGLVersion,
+                                               mozilla::WebGLVersion::WEBGL1,
+                                               mozilla::WebGLVersion::WEBGL2> {
+};
+
 } /* namespace IPC */
 
 namespace mozilla {
@@ -1232,7 +1240,7 @@
   }
 };
 
-} /* namespace ipc */
-} /* namespace mozilla */
+}  // namespace ipc
+}  // namespace mozilla
 
 #endif /* __GFXMESSAGEUTILS_H__ */